primitives/runtime/src/testing.rs

```diff
@@ -313,11 +313,17 @@ impl<Origin, Call, Extra> Applyable for TestXt<Call, Extra> where
 	/// Checks to see if this is a valid *transaction*. It returns information on it if so.
 	fn validate<U: ValidateUnsigned<Call=Self::Call>>(
 		&self,
-		_source: TransactionSource,
-		_info: &DispatchInfoOf<Self::Call>,
-		_len: usize,
+		source: TransactionSource,
+		info: &DispatchInfoOf<Self::Call>,
+		len: usize,
 	) -> TransactionValidity {
-		Ok(Default::default())
+		if let Some((ref id, ref extra)) = self.signature {
+			Extra::validate(extra, id, &self.call, info, len)
+		} else {
+			let valid = Extra::validate_unsigned(&self.call, info, len)?;
+			let unsigned_validation = U::validate_unsigned(source, &self.call)?;
+			Ok(valid.combine_with(unsigned_validation))
+		}
```

Comment:
this is the fix for the test, now `ValidateUnsigned::validate_unsigned` is called

Reply:
Can you make a test that ensures that this is called?

Reply:
yes done, an invalid extrinsic is now asserted to return 
```
Err(TransactionValidityError::Unknown(UnknownTransaction::NoUnsignedValidator)),
```